### PR TITLE
Fix blockseed for on_generated in mapgen env

### LINKED ISSUE #14868: 'blockseed' is the same for all mapchunks in the mapgen env
### Minetest version

```true
Minetest 5.9.0-dev-fcb4f258f (Linux)
Using LuaJIT 2.1.1699524327
```

### Irrlicht device

_No response_

### Operating system and version

Doesn't matter

### CPU model

_No response_

### GPU model

_No response_

### Active renderer

_No response_

### Summary

See title. Also see this IRC chat: https://irc.minetest.net/minetest/2024-07-19#i_6186246

### Steps to reproduce

Register a mapgen callback with `minetest.register_on_generated`; print the blockseed into the console.

Register a mapgen script with `minetest.register_mapgen_script`; print the blockseed into the console.

Observe that in the second case, the blockseed is the same for all chunks.

### PR DESCRIPTION
**Pull request by sfan5**
fixes #14868

### CODE REVIEW

File: src/mapgen/mapgen.h [hunk 1]
```diff
@@ -167,6 +167,7 @@ struct MapgenParams {
 */
 class Mapgen {
 public:
+	// Seed used for noises (truncated from the map seed)
 	s32 seed = 0;
```
Comment (JosiahWI) [inline src/mapgen/mapgen.h:171]: Would it be better to rename the variables to be more descriptive?

Comment (sfan5) [reply src/mapgen/mapgen.h:171]: Some day, maybe. But not now.

**Comment by SmallJoker** [comment]
Test script: main
```Lua
minetest.register_mapgen_script(minetest.get_modpath("lab") .. "/init_165_on_generate_log_thread.lua")

minetest.register_on_generated(function(minp, maxp, blockseed)
	print("main   thread: minp=" .. core.pos_to_string(minp) .. ": " .. blockseed)
end)
```
Test script: thread
```Lua
minetest.register_on_generated(function(vm, minp, maxp, blockseed)
	print("mapgen thread: minp=" .. core.pos_to_string(minp) .. ": " .. blockseed)
end)
```


Master:
```
mapgen thread: minp=(-272,-32,-32): 52534
main   thread: minp=(-272,-32,-32): 2267196001
mapgen thread: minp=(-272,-32,48): 52534
main   thread: minp=(-272,-32,48): 3616737801
mapgen thread: minp=(-352,-32,-32): 52534
main   thread: minp=(-352,-32,-32): 3087062625
mapgen thread: minp=(-352,-32,48): 52534
main   thread: minp=(-352,-32,48): 1706342729
mapgen thread: minp=(-352,-112,-32): 52534
main   thread: minp=(-352,-112,-32): 2413841841
mapgen thread: minp=(-352,-112,48): 52534
main   thread: minp=(-352,-112,48): 3967209409
mapgen thread: minp=(-352,48,-32): 52534
main   thread: minp=(-352,48,-32): 3957731241
mapgen thread: minp=(-352,48,48): 52534
main   thread: minp=(-352,48,48): 558002297
mapgen thread: minp=(-272,-112,-32): 52534
main   thread: minp=(-272,-112,-32): 1327296945
mapgen thread: minp=(-272,-112,48): 52534
main   thread: minp=(-272,-112,48): 3473049345
mapgen thread: minp=(-272,48,-32): 52534
main   thread: minp=(-272,48,-32): 2709552041
mapgen thread: minp=(-272,48,48): 52534
main   thread: minp=(-272,48,48): 1025171769
```

PR:
```
mapgen thread: minp=(-272,-32,-32): 2267196001
main   thread: minp=(-272,-32,-32): 2267196001
mapgen thread: minp=(-272,-32,48): 3616737801
main   thread: minp=(-272,-32,48): 3616737801
mapgen thread: minp=(-352,-32,-32): 3087062625
main   thread: minp=(-352,-32,-32): 3087062625
mapgen thread: minp=(-352,-32,48): 1706342729
main   thread: minp=(-352,-32,48): 1706342729
mapgen thread: minp=(-352,-112,-32): 2413841841
main   thread: minp=(-352,-112,-32): 2413841841
mapgen thread: minp=(-352,-112,48): 3967209409
main   thread: minp=(-352,-112,48): 3967209409
mapgen thread: minp=(-352,48,-32): 3957731241
main   thread: minp=(-352,48,-32): 3957731241
mapgen thread: minp=(-352,48,48): 558002297
main   thread: minp=(-352,48,48): 558002297
mapgen thread: minp=(-272,-112,-32): 1327296945
main   thread: minp=(-272,-112,-32): 1327296945
mapgen thread: minp=(-272,-112,48): 3473049345
main   thread: minp=(-272,-112,48): 3473049345
mapgen thread: minp=(-272,48,-32): 2709552041
main   thread: minp=(-272,48,-32): 2709552041
mapgen thread: minp=(-272,48,48): 1025171769
main   thread: minp=(-272,48,48): 1025171769
```

**Review by SmallJoker — APPROVED** [review]
Works.

**Pull request merged**